teuthology/orchestra/remote.py

```diff
@@ -448,7 +449,8 @@ def get_tar_stream(self, path, sudo=False):
     def os(self):
         if not hasattr(self, '_os'):
             try:
-                os_release = self.sh('cat /etc/os-release').strip()
+                os_release = self.sh('cat /etc/os-release',
+                                     stdout=StringIO()).strip()
```

Comment:
what???
remote.sh always returns str.

Reply:
Yeah, I saw the remote.sh code copied from teuthology, I thought the should be the same, but it is not.
In ceph/qa/vstart_runner it will return what it is.
Let me try to fix it in vstart_runner and to see whether is has any regression for the change.
Thanks.

Reply:
Tried to make the remote.sh to return str always in qa/tasks/vstart_runner.py and it works and didn't see any regression for other test cases.
I will fix it in ceph/qa code.

Reply:
@lxbsz which is that branch? why do we have a duplication of it vstart? :-)

Reply:
The vstart_runner.py will have its own remote class, which is a little different from the teuthology's. This is for the local test only for cephfs, more detail please see: https://docs.ceph.com/docs/master/dev/developer_guide/running-tests-locally/.

Reply:
The local test is a quick test case, I am new in this and I think many code in vstart_runner.py are duplicated to teuthology's we may try to reuse it.

Reply:
ah... this vstart black magic... I wonder why they duplicating code from teuthology? maybe it is time to get rid of it?

Reply:
I have no much idea about the history of the vstart_runner, and checked the code it seems we could reuse many code from teuthology, but not all.
For us we usually will do the local test first and then the integration test online, which is more efficient.